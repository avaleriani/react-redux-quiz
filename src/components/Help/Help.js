import React from "react";
import styles from "./styles.module.css";

/**
 * @types {{helpText: string, display: boolean}} props
 * @return HTMLElement
 */
const Btn = (props) => {
  const classes = [styles['help'], props.display ? styles['show'] : styles['hide']].join(' ');
  return (
    <span className={classes} role="img"
          aria-label="question-mark" data-tooltip={props.helpText}>‚ùì
  </span>
  );
};

export default Btn;
