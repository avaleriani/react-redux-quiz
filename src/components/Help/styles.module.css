.help {
    position: relative;
    cursor: pointer;
    top: -170px;
}

.help:after {
    opacity: 0;
    visibility: hidden;
    position: absolute;
    content: attr(data-tooltip);
    padding: 6px 10px;
    top: 1.4em;
    left: 50%;
    transform: translateX(-50%) translateY(-2px);
    background: grey;
    color: white;
    white-space: nowrap;
    z-index: 2;
    border-radius: 2px;
    transition: opacity 0.2s linear;
}

.help:hover::after {
    display: block;
    opacity: 1;
    visibility: visible;
    transform: translateX(-50%) translateY(0);
}

.show {
    visibility: visible;
}

.hide {
    visibility: hidden;
}