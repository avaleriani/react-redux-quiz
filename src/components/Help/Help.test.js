import React from 'react';
import renderer from 'react-test-renderer';
import Help from './Help';

describe('Help component', () => {

  test('renders correctly without props', () => {
    const HelpComponent = renderer.create(<Help/>).toJSON();
    expect(HelpComponent).toMatchSnapshot();
  });

  test('renders with props', () => {
    const props = { helpText: "test text", display: true };
    const HelpComponent = renderer.create(<Help {...props}/>).toJSON();
    expect(HelpComponent).toMatchSnapshot();
  });
});