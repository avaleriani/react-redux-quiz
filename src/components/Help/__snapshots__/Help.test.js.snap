// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Help component renders correctly without props 1`] = `
<span
  aria-label="question-mark"
  className="help hide"
  role="img"
>
  ❓
</span>
`;

exports[`Help component renders with props 1`] = `
<span
  aria-label="question-mark"
  className="help show"
  data-tooltip="test text"
  role="img"
>
  ❓
</span>
`;
